uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt5/client/paho/MqttConnectionImpl.java

```diff
@@ -92,9 +97,13 @@ public MqttSubscribeReply subscribe(long timeout, @NonNull List<Subscription> su
             mqttSubscriptions[i] = subscription;
             listeners[i] = new MqttMessageListener();
         }
+        MqttProperties properties = new MqttProperties();
+        if (userProperties != null && !userProperties.isEmpty()) {
+            properties.setUserProperties(convertUserPropertiesToList(userProperties));
+        }
         MqttSubscribeReply.Builder builder = MqttSubscribeReply.newBuilder();
         try {
-            IMqttToken response = client.subscribe(mqttSubscriptions, null, null, listeners, new MqttProperties());
+            IMqttToken response = client.subscribe(mqttSubscriptions, null, null, listeners, properties);
```

Comment:
Does client.subscribe() accept properties equals to null ?

Again: 
when no User properties set in the request it the same as NO info at all.
Empty hash not the same as NULL instead of empty object by semantic.

Reply:
I initialize properties in 100 lines as `MqttProperties properties = new MqttProperties();`, that's why It will never be null. We can use empty properties, because that's how other overloaded methods use it.